### ISSUE TITLE
Problem switching languages

### ISSUE THREAD

**Issue by ocrasborn** [Woseseltops/signbank#82]
If I change from Dutch to English by clicking on the flag (while in the search page), the language automatically switches back to Dutch after a few seconds.

**Comment by ocrasborn** [comment]
Hm. Can't reproduce this on my other computer. Will test some more.

**Comment by Woseseltops** [comment]
Hmm, weird. The language should only switch automatically from A to B in this particular situation:
1. You are logged in, and are using language B. This is saved in your user profile as 'last used language'.
2. You are logged out (later, on another device, or both), and are using language A.
3. You log in again.

**Comment by ocrasborn** [comment]
Hm. Can't reproduce this anymore (before I could on just one machine in Safari). Will put this to sleep for now.

**Comment by richardbank** [comment]
I can reproduce the above in Chrome on Win7 and Win8.1 (my default language is Dutch; when I click English, page reloads twice: first in English, then back to Dutch).
In FF and IE11, however, nothing happens when I click the English flag. Page reloads once (i think), but remains in Dutch.

**Comment by henrinie** [comment]
I tried this with our account, I can't change the language on Search page.
FireFox 38.2.1 and Red Hat Enterprise linux.

**Comment by ocrasborn** [comment]
Happens again to me now (Mac/Safari). Stored language is en; when I log in and change the language to Dutch on the search page, it reloads in English. I tried what happens if I change the stored language to nl in Django admin, but then I got to the error described in issue #86 ...

**Comment by Woseseltops** [comment]
I have made a discovery regarding this issue: it can be replicated by commenting out the request to `dictionary/switch_to_language/`, which is made when you click this flag while logged in to save the language in your user profile.

This then happens:
1. The website switches language, but this is not saved in the user profile
2. The website reloads to show the new language
3. Some Javascript added by me notices the discrepancy between the user profile and the current language, and fixes that.

In other words, it looks like your request to save the user preference is failing. This means you shouldn't have any problems when not logged in.

Furthermore, you guys note things not working in FF and IE at all, which I can reproduce. I suspect it never worked there, which is a separate issue. I'll make a another issue page for it.

**Comment by Woseseltops** [comment]
I think I fixed it in f2bed59fce8b6d3c2605e48bb41e9a2d629d9c8c . After I fixed the Firefox problem, I could reproduce the problem there. What happened was that the language switch was so fast that Firefox could not save the user preference, which caused the behavior I described above. The solution was to add a small delay. 

I think this will solve your problems as well, but I'm not fully sure of course.

**Comment by henrinie** [comment]
I tested it quickly. Language switching works on static pages, but not directly on the search page.

**Comment by Woseseltops** [comment]
Thanks, good find; it's fixed now. The problem wasn't static pages, but all pages that had another url than signbank.science.ru.nl (except for some corner cases). This was because the code to switch language added some thinks to the current url, so that if you were on http://signbank.science.ru.nl/dictionary/gloss/3117/ you would go to http://signbank.science.ru.nl/dictionary/gloss/3117/switch_to_language, which of course didn't work.
